src/coreclr/vm/methodtable.cpp

```diff
@@ -3443,6 +3443,15 @@ int MethodTable::GetLoongArch64PassStructInRegisterFlags(CORINFO_CLASS_HANDLE cl
                         size = STRUCT_FIRST_FIELD_SIZE_IS8;
                     }
                 }
+                else if (fieldType == ELEMENT_TYPE_CLASS)
+                {
+                    // Here is just skip the array as its elements' count greater than 1.
+                    // The `struct {int array[1]; float field_2;}` which the array field is only one element,
+                    // just skip it although it can be passed by registers one integer and one float-register.
+                    // Details see github https://github.com/dotnet/runtime/pull/62885#discussion_r821878981
```

Comment:
```suggestion
```
If you wanted to support inline fixed size arrays here, it would make most sense to base it on https://github.com/dotnet/csharplang/blob/main/proposals/inline-arrays.md . These inline arrays would show up as `ELEMENT_TYPE_VALUETYPE` here, not `ELEMENT_TYPE_CLASS`.

Reply:
Thanks very much.